### DISCUSSION TITLE
Dev Tools?

### DISCUSSION THREAD

**Discussion by omar-shahid** [xyflow/xyflow#2397] [Ideas]
Hi React Flow community,

I was wondering if there are any dev tools specific to react flow. It gets hard to know which node or edge we are looking at by just looking at the data. The React Flow works great when users have control of nodes and edges, but my requirements are stricter. I'm building a formatting algorithm that formats all nodes and edges in the grid. The user can only add new nodes or edit existing nodes.

 Currently, if I want to look for the exact node's data from the screen, I follow these steps:
- First, add its id in HTML
- From react dev tools, store all nodes data in a global variable
- Using the js' `find` method, I look for the node whose id is the required one.
> I can also search for it by position, but sometimes, especially when there is a bug, there can be hidden nodes overlapping each other or no new node, which makes this method less reliable.

My suggestion is a browser extension that can show each edge and node by clicking them, like regular HTML dev tools. A small version of this can be created right now in the existing react app, just by logging it into the console. 

Also the transitions of state of nodes and edges like we have in redux dev tools.

From looking in the docs, it seems like react flow is using `zustand`, which does have support for redux dev tools. 

So, basically, here's my suggestion for the dev tool:
- See each node and edge by clicking them
- History of nodes and edges, like redux
